src/redux/modules/evaluationProgress.js

```diff
@@ -85,14 +86,12 @@ export default handleActions(
             payload.submissions[0].webSocketChannel
           : payload.webSocketChannel;
 
-      return webSocketChannel
+      const newState = webSocketChannel
         ? initialState
-            .update('webSocketChannelId', () => webSocketChannel.id)
-            .update(
-              'expectedTasksCount',
-              () => webSocketChannel.expectedTasksCount
-            )
+            .set('webSocketChannelId', webSocketChannel.id)
+            .set('expectedTasksCount', webSocketChannel.expectedTasksCount)
         : initialState.set('isFinished', true);
+      return newState.set('progressObserverId', progressObserverId);
```

Comment:
The problem is rather complex. Multiple components on a page may re-use the Progress bar component used to visualize the progress of evaluation (i.e., updates passed via WebSocket from Monitor). Therefore, every entity that needs to do that must pass its observerId to the action that triggers the monitoring. The same ID is used in the ProgressBar dialog, so that only the right dialog is opened (and only one web socket connection is made). This ID needs to be in redux store since independent component on a page access it.

Reply:
And where do you get it... where is the original definition of `progressObserverId`?